platform/engineering/design-docs/2020-08-11-redirects-from-cms.md

```diff
@@ -0,0 +1,224 @@
+# Front End Build Redirects from CMS
+
+_Replace the previous line with the the title of your project or component and replace the following lines with your name(s), email and the date._
+**Author(s):** Steve Wirt
+**Last Updated:** Aug 21, 2020
+**Status:** Draft | **In Review** | Approved
+**Approvers:**
+   - [ ] Rian Fowler
+   - [ ] Andrew Gunsch
+   - [ ] Mikki Northuis
+   - [ ] Nick Sullivan
+   - [ ] Taylor Skinner (originally worked on redirect import)
+   - [ ] ??
+
+## Overview
+
+
+### Objective
+The CMS allows editors to create pages and move pages to other locations within the VA.gov website.  The CMS keeps track of when a page's path is changed and builds a redirect from one to the other.  The problem is, there is nothing that moves the redirects known within the CMS to the public website. It is in the best interest of editors, developers and veterans to have the front-end website know about and act on the redirects known to the CMS.
+
+### Background
+If pages are moved within the CMS under the right conditions the end result is a combination of internal broken links that currently cause failure of CMS content to be released to the front-end, and 404 errors for veterans.
+
+Within the CMS we have put in a lot of [safeguards to detect broken links](https://www.drupal.org/project/node_link_report) within content at the instant of saving a page of content and warning the edtitor in real time.  There are limits to this method however as it can not account for editors changing the location of a page being linked to or the published status of the link target at the time of content release (CR).
+
+| Situation | Current Result | Result if Redirects as Content |
+| --------- | -------------- | ------------------------------ |
+| Link correctly created in CMS to page in CMS that has been moved | 200 | 200 |
+| Link created by url copy paste in WYSIWG to page in CMS that has been moved | 404 & CR Fail | 200|
+| Link from VA.gov page not in CMS to CMS page that has been moved | 404 & CR Fail | 200 |
+| Link from external site to page in CMS that has been moved | 404 | 200 |
+| Browser bookmark / email link to page in CMS that has been moved | 404 | 200 |
+
+The current state of managing redirects for content, involves content editors and a developer coordinating and choreographing the release of the contnet with the addition of the redirect(s).  It is time intensive and risk prone.  VA needs a redirect system that is easier to manage and less reliant on developers timing releases perfectly to match.
+
+### Guiding Principles
+1. Veterans should not get 404 errors.
+2. Content on VA.gov should not have broken links to content also on VA.gov.
+3. As the CMS scales from ~20 editors and daily content releases, up to ~500 editors and continuous content releases, we can not allow a broken link to prevent content releases.
+4. Developers currently waste far too much time chasing down broken links.
```

Comment:
Thanks for adding these guiding principles! These also may be a good to start to defining what success looks like for a redirect strategy (which is more of a product question than the engineering design doc usually strives to answer, but still something I'm thinking about).